src/LiquidityPool.sol

```diff
@@ -249,11 +257,13 @@ contract LiquidityPool is Auth, IERC4626 {
     }
 
     /// @notice Request share redemption for a receiver to be included in the next epoch execution.
-    /// @notice Request can only be called by the owner of the shares
+    ///         DOES support flow where owner != msg.sender but has allowance to spend its shares
     ///         Shares are locked in the escrow on request submission
-    function requestRedeem(uint256 shares) public {
-        manager.requestRedeem(address(this), shares, msg.sender);
-        emit RedeemRequest(msg.sender, shares);
+    function requestRedeem(uint256 shares, address operator, address owner) public {
+        require(share.balanceOf(owner) >= shares, "LiquidityPool/insufficient-balance");
+        require(manager.requestRedeem(address(this), shares, operator), "LiquidityPool/request-redeem-failed");
+        require(transferFrom(owner, address(escrow), shares), "LiquidityPool/transfer-failed");
```

Comment:
we need to add a comment here that this is only possible because of the trusted forwarder pattern -> msg.sender is forwarded and the call can only be executed, if msg.sender has owner's approval to spent tokens